### PR TITLE
HttpMethodOverrider: simplify code

### PR DESCRIPTION
**Pull request by diesalbla**
Using `Alternative[Option]`, `as`, and `getOrElse`, is a rather convoluted way to say "if-then-else".

### CODE REVIEW

File: server/shared/src/main/scala/org/http4s/server/middleware/HttpMethodOverrider.scala [hunk 1]
```diff
@@ -147,15 +143,12 @@ object HttpMethodOverrider {
...
+        case None => http(req)
       }
 
     Kleisli { (req: Request[G]) =>
-      config.overridableMethods
-        .contains(req.method)
-        .guard[Option]
-        .as(processRequest(req))
-        .getOrElse(processRequestWithOriginalMethod(req))
+      val isOverridden = config.overridableMethods.contains(req.method)
```
Comment (danicheg) [inline server/shared/src/main/scala/org/http4s/server/middleware/HttpMethodOverrider.scala:150]: As a fan of `fold` on `Boolean`, I'd rather do it like:
```
import mouse.boolean._

config.overridableMethods.contains(req.method).fold(processRequest(req), http(req))
```
Much more concise than using `Alternative` and `map` + `getOrElse` (and better by perf). But we haven't `Mouse` for now :( Though your variant probably is the best by perf.

**Review by danicheg — APPROVED** [review]
Great. Could we have this for the `series/0.22`? Anyway, I can cherry-pick easily.

**Comment by armanbilge** [comment]
Do you mind if I try something?

**Comment by armanbilge** [comment]
@mergifyio backport series/0.22

**Comment by armanbilge** [comment]
Hmm, that's not very helpful :(

File: server/shared/src/main/scala/org/http4s/server/middleware/HttpMethodOverrider.scala (hunk 1 above)
Comment (diesalbla) [reply server/shared/src/main/scala/org/http4s/server/middleware/HttpMethodOverrider.scala:150]: In my humble opinion, it is best to stick to the language Scala itself, and to the `if-then-else` operator that is taught on week 2 of any introductory programming course. Using a `fold` for `Boolean` seems to stretch that word a lot.

Comment (rossabaker) [reply server/shared/src/main/scala/org/http4s/server/middleware/HttpMethodOverrider.scala:150]: Also not a fan of boolean fold.  Makes me pause and think whether true or false comes first, and my guess was wrong.

**Comment by rossabaker** [comment]
Cherry-picked to 0.22.

**Pull request closed without merging**